### PR TITLE
feat: create a slide switching directive #1162

### PR DESCRIPTION
**Pull request by MaxFlower**
This is a draft where I propose to implement only one directive with entry param which sets the direction of switching between slides (`next` or `privious`). @kirjs what do you think?

### CODE REVIEW

File: apps/codelab/src/app/components/exercise/exercise.component.html [hunk 1]
```diff
@@ -26,7 +26,9 @@
         [code]="code"
       ></codelab-babel-test-runner>
       <div *ngIf="isSolved" class="solved-message">
-        <h3 i18n>Well done! Now you can go to the next slide</h3>
+        <h3 i18n switchSlide="next">
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/components/exercise/exercise.component.html]: I'd just go with two directives - `nextSlide` and `previousSlide`.

**Comment by MaxFlower** [comment]
Turned switchSlide directive into 2 separate directives.

File: apps/codelab/src/app/components/exercise/exercise.component.css [hunk 2]
```diff
@@ -17,6 +17,12 @@
   margin-top: 16px;
 }
 
+.solved-message h3:hover {
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/components/exercise/exercise.component.css]: Now that it's clickable can we make it  accessible?
E.g. make it a `button`?

Comment (MaxFlower) [reply apps/codelab/src/app/components/exercise/exercise.component.css]: Sure. I thought about it.

Comment (MaxFlower) [reply apps/codelab/src/app/components/exercise/exercise.component.css]: Done.

File: apps/codelab/src/app/directives/nextSlide.directive.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Directive, HostListener } from '@angular/core';
+import { SlidesDeckComponent } from '@codelab/slides/src/lib/deck/deck.component';
+
+@Directive({
+  selector: 'button[nextSlide]'
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/directives/nextSlide.directive.ts]: I don't think we should be restricting this directive to a button tag, there are many other use cases/ways to make it accessible

Comment (MaxFlower) [reply apps/codelab/src/app/directives/nextSlide.directive.ts]: Actually I particaly agree with that but any way I removed that restriction.

**Pull request merged**